src/catalog/services/df_service.py

```diff
@@ -87,3 +87,10 @@ def dataset_object_by_name(self, database_name, dataset_name,
         """
         return self.model.query.filter(
             self.model._name == dataset_name).one()
+        
+    def delete_dataset(self, metadata_id):
+        result = self.model.query \
+            .filter(self.model._id == metadata_id) \
+            .one()
+
```

Comment:
Use try-catch here. We will be refactoring other places as-well where we are missing it.